waterdata/views.py

```diff
@@ -33,51 +33,77 @@ def monitoring_location(site_no):
 
     resp = execute_get_request(SERVICE_ROOT,
                                path='/nwis/site/',
-                               params={'site': site_no,
-                                       'agencyCd': agency_cd,
+                               params={'site'      : site_no,
+                                       'agencyCd'  : agency_cd,
                                        'siteOutput': 'expanded',
-                                       'format': 'rdb'
-                                       }
-                               )
+                                       'format'    : 'rdb'
+                                      }
+                              )
     status = resp.status_code
     if status == 200:
         iter_data = parse_rdb(resp.iter_lines(decode_unicode=True))
-        station_record = next(iter_data)
-        parameter_data_resp = execute_get_request(SERVICE_ROOT,
-                                                  path='/nwis/site/',
-                                                  params={'format': 'rdb',
-                                                          'sites': site_no,
-                                                          'seriesCatalogOutput': True,
-                                                          'siteStatus': 'all',
-                                                          'agencyCd': agency_cd
-                                                          }
-                                                  )
-        if parameter_data_resp.status_code == 200:
-            param_data = parse_rdb(parameter_data_resp.iter_lines(decode_unicode=True))
-            location_capabilities = get_capabilities(param_data)
-            json_ld = build_linked_data(site_no,
-                                        station_record.get('station_nm'),
-                                        station_record.get('agency_cd'),
-                                        station_record.get('dec_lat_va', ''),
-                                        station_record.get('dec_long_va', ''),
-                                        location_capabilities
-                                        )
-            safe_json_ld = Markup(json.dumps(json_ld, indent=4))
-        else:
-            safe_json_ld = None
+
+        data_list = list(iter_data)
+
         template = 'monitoring_location.html'
-        context = {
-            'status_code'       : status,
-            'station'           : station_record,
-            'location_with_values' : get_disambiguated_values(
-                station_record,
-                app.config['NWIS_CODE_LOOKUP'],
-                app.config['COUNTRY_STATE_COUNTY_LOOKUP']
-            ),
-            'STATION_FIELDS_D'  : STATION_FIELDS_D,
-            'json_ld': safe_json_ld
-        }
+
+        station_nm = ''
+        site_no = ''
+        if len(data_list) == 1:
```

Comment:
This should be reworked so that this if clause is only made once (they other is on line 63).